src/main/java/com/airbnb/airpal/AirpalApplicationBase.java

```diff
@@ -0,0 +1,95 @@
+package com.airbnb.airpal;
+
+import com.airbnb.airpal.core.AirpalUserFactory;
+import com.airbnb.airpal.core.health.PrestoHealthCheck;
+import com.airbnb.airpal.modules.AirpalModule;
+import com.airbnb.airpal.modules.DropwizardModule;
+import com.airbnb.airpal.resources.ExecuteResource;
+import com.airbnb.airpal.resources.FilesResource;
+import com.airbnb.airpal.resources.HealthResource;
+import com.airbnb.airpal.resources.PingResource;
+import com.airbnb.airpal.resources.QueriesResource;
+import com.airbnb.airpal.resources.QueryResource;
+import com.airbnb.airpal.resources.ResultsPreviewResource;
+import com.airbnb.airpal.resources.S3FilesResource;
+import com.airbnb.airpal.resources.SessionResource;
+import com.airbnb.airpal.resources.TablesResource;
+import com.airbnb.airpal.resources.UserResource;
+import com.airbnb.airpal.resources.UsersResource;
+import com.airbnb.airpal.resources.sse.SSEEventSourceServlet;
+import com.google.inject.Injector;
+import io.dropwizard.Application;
+import io.dropwizard.assets.AssetsBundle;
+import io.dropwizard.db.DataSourceFactory;
+import io.dropwizard.flyway.FlywayBundle;
+import io.dropwizard.flyway.FlywayFactory;
+import io.dropwizard.setup.Bootstrap;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.views.ViewBundle;
+import org.secnod.dropwizard.shiro.ShiroBundle;
+import org.secnod.dropwizard.shiro.ShiroConfiguration;
+
+import javax.servlet.ServletRegistration;
+
+import static com.sun.jersey.core.util.ReaderWriter.BUFFER_SIZE_SYSTEM_PROPERTY;
+
+public abstract class AirpalApplicationBase<T extends AirpalConfiguration> extends Application<T>
+{
+    protected void initializeBase(Bootstrap<T> bootstrap) {
+        final AssetsBundle assetBundle = new AssetsBundle("/assets", "/app", "index.html");
+        final ViewBundle viewBundle = new ViewBundle();
+        final FlywayBundle<AirpalConfiguration> flywayBundle = new FlywayBundle<AirpalConfiguration>() {
```

Comment:
This and all references to `AirpalConfiguration` below should be changed to reference our new type parameter.